### PR TITLE
Loop times revamp

### PR DESCRIPTION
**Pull request by PikaDude**
This PR introduces the following:
- An upgraded loop times module that now supports weather
- A loop times debug function, that will skim the song to right before the loop. Used to test the loop
- Loop times for every single hourly song (all 192). These loop times now allow for seamless song looping, skipping any intros or outros
- A fix causing the loop timer to start before the song actually begins playing
- Optimisations and fixes that don't make the loop timer reload the song every loop
- A small fix for the `formatHour` function, making the checks non-strict, so passing strings will now result in an accurate response

### CODE REVIEW

**Comment by jordigarcl** [comment]
I've been postponing reviewing this PR as it would require some time. I should be able to allocate time tomorrow!

**Comment by jordigarcl** [comment]
Will check this now

File: scripts/background/Utility.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ var DEBUG_FLAG = false;
 
 // format text for the badge and for the song URL
 function formatHour(time) {
-	if (time === -1) {
+	if (time == -1) {
```
Comment (jordigarcl) [inline scripts/background/Utility.js:9]: What's the reason for this change? I guess we are sending `time` as a string somewhere?  Damn javascript and its equality checks

**Review by jordigarcl — APPROVED** [review]

File: scripts/background/Utility.js (hunk 1 above)
Comment (PikaDude) [reply scripts/background/Utility.js:9]: This was an issue that occurred during testing, it's not really needed, however was done to prevent any future issues as strict equality checks should only be used when necessary.
